inc/class-sitemaps-posts.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+/**
+ * Class Core_Sitemaps_Posts.
+ * Builds the sitemap pages for Posts.
+ */
+class Core_Sitemaps_Posts {
+	/**
+	 * @var Core_Sitemaps_Registry object
+	 */
+	public $registry;
+
+	/**
+	 * Core_Sitemaps_Index constructor.
+	 */
+	public function __construct() {
+		$this->registry = Core_Sitemaps_Registry::instance();
+	}
+
+	/**
+	 * Bootstrapping the filters.
+	 */
+	public function bootstrap() {
+		add_action( 'core_sitemaps_setup_sitemaps', array( $this, 'register_sitemap' ), 99 );
+		add_filter( 'template_redirect', array( $this, 'render_sitemap' ) );
+	}
+
+	/**
+	 * Sets up rewrite rule for sitemap_index.
+	 */
+	public function register_sitemap() {
+		$this->registry->add_sitemap( 'posts', '^sitemap-posts\.xml$' );
```

Comment:
Do we also need to store a date modified here as well? On the sitemaps index we need to display the `<loc>` and `<lastmod>` values

Reply:
Eventually, we'll need to be able to get the lastmod from each sitemap page, but let's not worry about that here. For now, just being able to see the sitemap rendered out will be fine.

Reply:
There's no link here or mechanism to the page object to get the last modified date from (either way) so currently that's not possible.